ZenPacks/zenoss/Microsoft/Windows/modeler/plugins/zenoss/winrm/WinMSSQL.py

```diff
@@ -229,126 +234,154 @@ def __init__(self, conn_info, log):
+        "$res['ao_enabled'] = $server.IsHadrEnabled;"
+        "$ags = New-Object System.Collections.ArrayList;"
+        "$ars = New-Object System.Collections.ArrayList;"
+        "$adbs = New-Object System.Collections.ArrayList;"
+
+        "if ($server.IsHadrEnabled -and $server.AvailabilityGroups.Length -gt 0) {"
+        " $epoch = [timezone]::CurrentTimeZone.ToLocalTime([datetime]'1/1/1970').ToUniversalTime();"
+        " $dbs = $server.Databases;"
+        " $ag_res_ids_map = New-Object 'system.collections.generic.dictionary[string, object]';"
+        " $inst_to_node_map = New-Object 'system.collections.generic.dictionary[string, object]';"
+        " $server.Refresh();"
+        " $res['sql_server_fullname'] = $server.DomainInstanceName;"
+        " $res['is_clustered'] = $server.IsClustered;"
+        " $res['is_on_wsfc'] = $server.IsMemberOfWsfcCluster;"
+
+        " $dbmaster = $server.Databases['master'];"
+
+        " $cluster_query = \\\"SELECT ag_id, ag_resource_id FROM sys.dm_hadr_name_id_map;"
+        " SELECT instance_name, node_name FROM sys.dm_hadr_instance_node_map;"
+        " SELECT member_name, member_type, member_state, number_of_quorum_votes FROM sys.dm_hadr_cluster_members;\\\";"
+        " $cl_qry_res = $dbmaster.ExecuteWithResults($cluster_query);"
+        " try {"
+        "  foreach ($row in $cl_qry_res.tables[0].rows) {"
+        "   $ag_res_ids_map[$row.ag_id] = $row.ag_resource_id;"
+        "  }"
+        "  foreach ($row in $cl_qry_res.tables[1].rows) {"
+        "   $inst_to_node_map[$row.instance_name] = $row.node_name;"
+        "  }"
+        " } catch {}"
+        " try {"
+        "  $cl_members = New-Object System.Collections.ArrayList;"
+        "  foreach ($row in $cl_qry_res.tables[2].rows) {"
+        "   $cl_member = New-Object 'system.collections.generic.dictionary[string, object]';"
+        "   $cl_member['member_name'] = $row.member_name;"
+        "   $cl_member['member_type'] = $row.member_type;"
+        "   $cl_member['member_state'] = $row.member_state;"
+        "   $cl_member['number_of_quorum_votes'] = $row.number_of_quorum_votes;"
+        "   $cl_members.Add($cl_member) > $null;"
+        "  }"
+        "  $res['wsfc_cluster_members'] = $cl_members;"
+        " }"
+        " catch {}"
+        " foreach ($ag in $server.AvailabilityGroups) {"
+        "  $ag_info = New-Object 'system.collections.generic.dictionary[string, object]';"
+        "  $ag_uid = $ag.UniqueId;"
+        "  $ag_res_uid = $ag_res_ids_map[$ag_uid];"
+        "  $pr_rep_servr_name = $ag.PrimaryReplicaServerName;"
+        "  $ag_info['name'] = $ag.Name;"
+        "  $ag_info['id'] = $ag_uid;"
+        "  $ag_info['ag_res_id'] = $ag_res_uid;"
+        "  $ag_info['primary_replica_server_name'] = $pr_rep_servr_name;"
+        # Put databases on top beacause we get them regardless of Replica role (either on Primary or Secondary Replicas)
+        "  foreach ($adb in $ag.AvailabilityDatabases) {"
```

Comment:
Have added database section. The rest is only script length optimization by string refactoring.